crates/bevy_render/src/camera/camera.rs

```diff
@@ -1,46 +1,115 @@
-use std::marker::PhantomData;
-
 use crate::{
     camera::CameraProjection,
     prelude::Image,
     render_asset::RenderAssets,
     render_resource::TextureView,
     view::{ExtractedView, ExtractedWindows, VisibleEntities},
-    RenderApp, RenderStage,
 };
-use bevy_app::{App, CoreStage, Plugin, StartupStage};
 use bevy_asset::{AssetEvent, Assets, Handle};
+use bevy_derive::{Deref, DerefMut};
 use bevy_ecs::{
     change_detection::DetectChanges,
     component::Component,
     entity::Entity,
     event::EventReader,
-    prelude::With,
     query::Added,
     reflect::ReflectComponent,
-    system::{Commands, ParamSet, Query, Res, ResMut},
+    system::{Commands, ParamSet, Query, Res},
 };
 use bevy_math::{Mat4, UVec2, Vec2, Vec3};
 use bevy_reflect::prelude::*;
 use bevy_transform::components::GlobalTransform;
 use bevy_utils::HashSet;
 use bevy_window::{WindowCreated, WindowId, WindowResized, Windows};
 use serde::{Deserialize, Serialize};
+use std::borrow::Cow;
 use wgpu::Extent3d;
 
-#[derive(Component, Default, Debug, Reflect, Clone)]
-#[reflect(Component, Default)]
+#[derive(Component, Debug, Reflect, Clone)]
+#[reflect(Component)]
 pub struct Camera {
     pub projection_matrix: Mat4,
+    pub logical_target_size: Option<Vec2>,
+    pub physical_target_size: Option<UVec2>,
+    pub priority: isize,
+    pub is_active: bool,
     #[reflect(ignore)]
     pub target: RenderTarget,
     #[reflect(ignore)]
     pub depth_calculation: DepthCalculation,
-    pub near: f32,
-    pub far: f32,
 }
 
-#[derive(Debug, Clone, Reflect, PartialEq, Eq, Hash)]
+impl Default for Camera {
+    fn default() -> Self {
+        Self {
+            is_active: true,
+            priority: 0,
+            projection_matrix: Default::default(),
+            logical_target_size: Default::default(),
+            physical_target_size: Default::default(),
+            target: Default::default(),
+            depth_calculation: Default::default(),
+        }
+    }
+}
+
+impl Camera {
+    /// Given a position in world space, use the camera to compute the screen space coordinates.
+    ///
+    /// To get the coordinates in Normalized Device Coordinates, you should use
+    /// [`world_to_ndc`](Self::world_to_ndc).
+    pub fn world_to_screen(
+        &self,
+        camera_transform: &GlobalTransform,
+        world_position: Vec3,
+    ) -> Option<Vec2> {
```

Comment:
I think the resulting coordinate is not anymore in screen space (from the full window), but rather in "view space" ?

It might be useful to keep previous `world_to_screen` or a new `world_to_window_screen` method ?

Reply:
Nope, if world_to_ndc does indeed output ndc, then this outputs screen coordinates.